src/WebJobs.Extensions.MobileApps/Config/MobileAppsConfiguration.cs

```diff
@@ -53,8 +62,194 @@ public void Initialize(ExtensionConfigContext context)
                 throw new ArgumentNullException("context");
             }
 
-            context.Config.RegisterBindingExtension(
-                new MobileTableAttributeBindingProvider(context.Config, this, context.Config.NameResolver));
+            INameResolver nameResolver = context.Config.GetService<INameResolver>();
+            IConverterManager converterManager = context.Config.GetService<IConverterManager>();
+
+            BindingFactory factory = new BindingFactory(nameResolver, converterManager);
+
+            IBindingProvider outputProvider = factory.BindToGenericAsyncCollector<MobileTableAttribute, MobileTableContext>(typeof(MobileTableAsyncCollector<>), BindForOutput);
+
+            IBindingProvider clientProvider = factory.BindToExactType<MobileTableAttribute, IMobileServiceClient>(BindForClient);
+
+            IBindingProvider queryProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForQuery);
+            queryProvider = factory.AddTypeFilter(IsQueryType, queryProvider);
+            queryProvider = factory.AddValidator<MobileTableAttribute>(ThrowIfGenericArgumentIsInvalid, queryProvider);
+
+            IBindingProvider jObjectTableProvider = factory.BindToExactType<MobileTableAttribute, IMobileServiceTable>(BindForTable);
+
+            IBindingProvider tableProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForTable);
+            tableProvider = factory.AddTypeFilter(IsTableType, tableProvider);
+            tableProvider = factory.AddValidator<MobileTableAttribute>(ThrowIfGenericArgumentIsInvalid, tableProvider);
+
+            IBindingProvider itemProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForItem);
+            itemProvider = factory.AddValidator<MobileTableAttribute>(IsItemType, itemProvider);
+
+            IExtensionRegistry extensions = context.Config.GetService<IExtensionRegistry>();
+            extensions.RegisterBindingRules<MobileTableAttribute>(ValidateMobileAppUri, nameResolver, outputProvider, clientProvider, jObjectTableProvider, queryProvider, tableProvider, itemProvider);
+        }
+
+        internal bool IsQueryType(Type paramType)
```

Comment:
Yeah, I think we should promote a clean class separation pattern. People will end up doing that themselves anyways to keep things straight, rather than a mega class with a bunch of little possibly unrelated funcs.